distribution/openrct2.d.ts

```diff
@@ -265,10 +265,10 @@ declare global {
          * @param execute Logic for validating and executing the action.
          * @throws An error if the action has already been registered by this or another plugin.
          */
-        registerAction(
+        registerAction<T = object>(
```

Comment:
I do not see the advantage of having a generic here.

Reply:
Currently you need to do an unsafe cast in all callbacks from `object` to `any` or an interface that has your parameters. This generic automatically accepts whatever interface you wanna use for your arguments, and it ensures both the query and execute callbacks use the same interface.

See [this Typescript playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgOJwLYQIILMAexACUIBnAVwBsxkBvAWACgBIaKAqAfgC5kQKGAEbQA3MzZQOUACrAwVCL2RkwUUAHNxrdpwCy5MnA1K+q9SC0SEBVcoHCxzAL7NmoSLEQp0WXPiIAUQA3CHBsKA0yAB4ZZABeZAIhACsIPAA+egkoCDgAEyIqAE9kAAcqOGLoPgcRKG0WXIKi0rBisohawXrG5sKQEuREAJAzNU0+vIGh4DIAYSpgMLAAeUHiviECAkU4ECmWjeHIsj4ZKcoaPl8cPEIScmowbVcmd3BoeCRkAAUqCgaUARKLZViQVQARm6jgaEghYAATOMLBoXG4mDYQKpkAg4FQqEJEABrBLIAAUEFC4BumDuoxCKxBMX+gOBpwyAEoEhlmIxWFiyLsIAA6KgEDSU6lgEVwU4ihGQgA0yCpK1l8oRiM5UzAFCgIHozleGJgFBA9yIyFyQNU0H8D1QYWgwAQsTJyTSmXJEhGDxRmiVEgAjhRoJsKXKorS-JaQIzwqdYlyeWg6Q6iKQrmAgzoAB7pCiQPjkqNnNOxhnS5nJ7nxLK3DOPbOcvjBAjAfJ8t42uaeJtOkAuhDkgBECNHKrxBKJCGJU-xhJJOuQAHpV8gAO6cYlkLfyAAWwyLBAwcHwCFxcFUmmGIHyqux+tv2zAR+nS7ne4PcFCyDfKBGFg-4dBAprmnG1oQLa-ZxgAckQg7Dj6rB+kQAaWLmLChuGJZlnwnrpGAdYNumcZZs8WEQAWCBFl0kanARqRESRFb0g8FE0K2yDtp23bML2dpQE2CEgEh6gjuO5BgJOV4ziSC7yXOK7rsghTkPwBC0NuUDEkAA) showcasing this problem.

Reply:
I see.